### PR TITLE
Incorporate pastro into online deployment

### LINKED ISSUE #250: Implement `p_astro` in online deployment
We should incorporate the new `p_astro` module into our online deployment

### PR DESCRIPTION
**Pull request by EthanMarx**
Closes #250

### CODE REVIEW

File: projects/online/online/utils/gdb.py [hunk 1]
```diff
@@ -78,6 +79,23 @@ def submit_pe(
...
+        pastro = {
+            "BBH": pastro,
+            "Terrestrial": 1 - pastro,
+            "NSBH": 0,
+            "BNS": 0,
+        }
+        with open(fname, "w") as f:
+            json.dump(pastro, f)
+
+        self.writeLog(
```
Comment (wbenoit26) [inline projects/online/online/utils/gdb.py:95]: Can change this to snake case

**Review by wbenoit26 — CHANGES_REQUESTED** [review]

**Comment by EthanMarx** [comment]
@wbenoit26 Huge amount of file changes here b/c I added all of our local libraries as "first parties" to `isort` so that their imports will be grouped together. Let me try to cherry pick these commits into a separate PR

File: projects/online/online/main.py [hunk 2]
```diff
@@ -97,8 +108,15 @@ def process_event(
         outdir / "whitened_data_plots",
         device,
     )
+
+    # submit the posterior and skymap to gracedb
+    # using the graceid from the event submission
     graceid = response.json()["graceid"]
     gdb.submit_pe(posterior, skymap, graceid)
+
+    # calculate and submit pastro
```
[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/online/online/main.py:117]: Yeah so we should definitely multiprocess this, but for public alerts the only thing that matters is FAR and skymap which is why I chose those first

[the remark this answers is not in the document]
Comment (wbenoit26) [reply projects/online/online/main.py:117]: Got it, the order makes sense then. Can you open up an issue to multiprocess these outputs?

File: projects/online/online/main.py [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
 from architectures import Architecture
-from ledger.events import EventSet
+from ledger.events import EventSet, RecoveredInjectionSet
+from ledger.injections import InjectionParameterSet
 from ligo.gracedb.rest import GraceDb
-from ml4gw.transforms import ChannelWiseScaler, SpectralDensity, Whiten
 from online.utils.buffer import InputBuffer, OutputBuffer
 from online.utils.dataloading import data_iterator
 from online.utils.gdb import gracedb_factory
+from online.utils.pastro import fit_or_load_pastro
```
[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/online/online/main.py:15]: just added it

File: projects/online/online/utils/gdb.py (hunk 1 above)
Comment (EthanMarx) [reply projects/online/online/utils/gdb.py]: done

**Comment by EthanMarx** [comment]
@wbenoit26 think this is ready now

**Comment by wbenoit26** [comment]
Looks like you need to add the `p_astro` lib to `apptainer.def`. Otherwise, yeah, this looks good

**Comment by EthanMarx** [comment]
@wbenoit26  Made `is_sorted_by` a "public" method, and added sorting tests for the generic `ledger.Ledger` object before i realized you already did this in `test_events.py` but the more tests the merrier :)

**Comment by wbenoit26** [comment]
Hah, excellent, I should've put some tests there to begin with, so that's good. 

Also, looks like a notebook snuck in.

**Comment by wbenoit26** [comment]
Maybe we should add `**/*.ipynb` to `.git_ignore`

**Comment by EthanMarx** [comment]
Yeah probably a good idea - if we ever find a reason to add ipynbs we can remove it

**Review by wbenoit26 — APPROVED** [review]

**Pull request merged**